addons/web_tour/static/src/js/tour_service.js

```diff
@@ -140,18 +148,41 @@ return session.is_bound.then(function () {
         });
 
         // Now that the observer is configured, we have to start it when needed.
+        const observerOptions = {
+            attributes: true,
+            childList: true,
+            subtree: true,
+            attributeOldValue: true,
+            characterData: true,
+        };
+
         var start_service = (function () {
             return function (observe) {
                 return new Promise(function (resolve, reject) {
                     tour_manager._register_all(observe).then(function () {
                         if (observe) {
-                            observer.observe(document.body, {
-                                attributes: true,
-                                childList: true,
-                                subtree: true,
-                                attributeOldValue: true,
-                                characterData: true,
+                            observer.observe(document.body, observerOptions);
+
+                            // If an iframe is added during the tour, its DOM
+                            // mutations should also be observed to update the
+                            // tour manager.
+                            const containsOrIsIframe = node => {
+                                const isElement = node.nodeType === 1;
+                                return isElement && (node.querySelector('.o_iframe') || node.matches('.o_iframe'));
+                            };
+                            const iframeObserver = new MutationObserver(mutations => {
+                                const iframeMutation = mutations.filter(mutation => Array.from(mutation.addedNodes).some(containsOrIsIframe))[0];
+                                if (iframeMutation) {
+                                    let iframeEl = iframeMutation.target.querySelector('.o_iframe');
+                                    if (iframeMutation.target.matches('.o_iframe')) {
+                                        iframeEl = iframeMutation.target;
+                                    }
```

Comment:
Same: why `querySelector` if you already have an iframe, strange to read and probably not as fast.